runtime/domain.c

```diff
@@ -906,6 +899,7 @@ static void* domain_thread_func(void* v)
     caml_gc_log("Domain starting (unique_id = %"ARCH_INTNAT_PRINTF_FORMAT"u)",
                 domain_self->interruptor.unique_id);
     caml_domain_set_name("Domain");
+    CAML_EV_LIFECYCLE(EV_DOMAIN_SPAWN, getpid());
```

Comment:
Why is the argument `getpid()` here, and also on domain termination? I would rather expect the current domain id? (Should it be `domain_state->id` or `domain_state->unique_id`?)

Reply:
The domain id on which an event occurred is always passed to callbacks. For `EV_DOMAIN_SPAWN` we pass the pid as data to go along with that. Consumers can then associate that with the domain id and potentially use it to gather other system data (e.g scraping /proc for disk or network IO data).